contracts/SDIncentiveController.sol

```diff
@@ -0,0 +1,146 @@
+        if (reward == 0) {
+            revert NoRewardsToClaim();
+        }
+        rewards[account] = 0;
+        IERC20(staderConfig.getStaderToken()).transfer(account, reward);
+
+        emit RewardClaimed(account, reward);
+    }
+
+    /// @notice Updates the reward on deposit in the utility pool.
+    /// @param account The account that made a deposit.
+    function onDelegate(address account) external override {
+        UtilLib.onlyStaderContract(msg.sender, staderConfig, staderConfig.SD_UTILITY_POOL());
+
+        updateReward(account);
+    }
+
+    /// @notice Updates the address of staderConfig
+    /// @param _staderConfig The new address of staderConfig
+    function updateStaderConfig(address _staderConfig) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        UtilLib.checkNonZeroAddress(_staderConfig);
+        staderConfig = IStaderConfig(_staderConfig);
+        emit UpdatedStaderConfig(_staderConfig);
+    }
+
+    /// @notice Updates the emission rate of the reward tokens per block.
+    /// @param newEmissionRate The new emission rate per block.
+    function updateEmissionRate(uint256 newEmissionRate) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (newEmissionRate == 0) revert InvalidEmissionRate();
+        emissionPerBlock = newEmissionRate;
+        emit EmissionRateUpdated(newEmissionRate);
+    }
+
+    /// @notice Updates the end block of the reward period.
+    /// @param _newEndBlock The new end block.
+    function updateEndBlock(uint256 _newEndBlock) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        if (_newEndBlock <= block.number) revert InvalidEndBlock();
+        rewardEndBlock = _newEndBlock;
+        emit RewardEndBlockUpdated(_newEndBlock);
+    }
+
+    /// @notice Calculates the current reward per token.
+    /// @return The calculated reward per token.
+    function rewardPerToken() public view override returns (uint256) {
+        if (block.number >= rewardEndBlock) {
+            return rewardPerTokenStored;
+        }
+
+        uint256 totalSupply = ISDUtilityPool(staderConfig.getSDUtilityPool()).cTokenTotalSupply();
+        if (totalSupply == 0) {
+            return rewardPerTokenStored;
+        }
+        return
+            rewardPerTokenStored +
+            (((block.number - lastUpdateBlockNumber) * emissionPerBlock * DECIMAL) / totalSupply);
+    }
+
+    /// @notice Calculates the total accrued reward for an account.
+    /// @param account The account to calculate rewards for.
+    /// @return The total accrued reward for the account.
+    function earned(address account) public view override returns (uint256) {
+        uint256 currentBalance = ISDUtilityPool(staderConfig.getSDUtilityPool()).delegatorCTokenBalance(account);
+        uint256 currentRewardPerToken = rewardPerToken();
+
+        return
+            ((currentBalance * (currentRewardPerToken - userRewardPerTokenPaid[account])) / DECIMAL) + rewards[account];
```

Comment:
Don't think it hurts here. But anytime, we find value of userRewardPerTokenPaid to be 0, we should assume it to be equal to rewardPerToken(), right?